src/arrayMethodLastIndexOf.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomLastIndexOf() {
   [].__proto__.lastIndexOf2 = function(searchElement, fromIndex) {
-    // write code here
+    if (this.length === 0) {
+      return -1;
+    }
+    let fromIndexInt = fromIndex;
+    if (typeof fromIndex === 'undefined') {
+      fromIndexInt = this.length - 1;
+    } else if (fromIndexInt < 0) {
+      fromIndexInt = this.length + fromIndex;
+      if (fromIndexInt < 0) {
+        return -1;
+      }
+    }
+
+    let i = fromIndexInt;
+    while (true) {
+      if (this[i] === searchElement) {
+        return i;
+      }
+      i--;
+      if (i < 0) {
+        break;
```

Comment:
`break`, `continue` is bad practice in js. Most times you can replace them with return.
Also, will be better to use for loop, when you have finite counter instead of manually manage counter with while(true).